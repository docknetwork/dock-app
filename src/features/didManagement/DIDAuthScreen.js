import {authHandler} from '../qr-code-scanner/qr-code';
import {useIsFocused} from '@react-navigation/native';
import React, {useCallback, useEffect, useState} from 'react';
import {ScreenContainer, Typography} from '../../design-system';
import {Spinner} from 'native-base';
import {translate} from '../../locales';
import {navigate} from '../../core/navigation';
import {Routes} from '../../core/routes';

export function DIDAuthScreen({authState}) {
  return (
    <ScreenContainer testID="DIDAuthScreen">
      {authState === 'processing' ? <Spinner /> : null}
      <Typography variant="description" marginLeft={2}>
        {translate(`qr_scanner.${authState}`)}
      </Typography>
    </ScreenContainer>
  );
}
export function DIDAuthScreenContainer({route}) {
  const {dockWalletAuthDeepLink} = route.params || {};
  const isScreenFocus = useIsFocused();
  const [authState, setAuthState] = useState('start');

  const authenticateDid = useCallback(async () => {
    setAuthState('processing');
    await authHandler(dockWalletAuthDeepLink);
    setAuthState('completed');
    setTimeout(() => {
      navigate(Routes.ACCOUNTS);
    }, 2000);
  }, [dockWalletAuthDeepLink]);

  useEffect(() => {
    if (dockWalletAuthDeepLink && isScreenFocus) {
      authenticateDid();
    }
  }, [authenticateDid, dockWalletAuthDeepLink, isScreenFocus]);
  return <DIDAuthScreen authState={authState} />;
}
