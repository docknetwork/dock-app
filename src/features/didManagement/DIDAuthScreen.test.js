import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {DIDAuthScreen} from './DIDAuthScreen';

const mockStore = configureMockStore();

describe('QRScanScreen', () => {
  it('should render correctly when isScreenFocus=true', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(<DIDAuthScreen authState={'start'} />, {
      context: {store: mockStore(initialState)},
    });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
