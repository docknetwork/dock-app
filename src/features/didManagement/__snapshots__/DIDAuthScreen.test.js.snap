// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRScanScreen should render correctly when isScreenFocus=true 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#000",
      "flex": 1,
    }
  }
  testID="DIDAuthScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(Component)
        marginLeft={2}
        variant="description"
      >
        Start
      </Styled(Component)>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;
